app/Packages/TaskFighter/routes.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+use Illuminate\Http\Response;
+
+/*
+|--------------------------------------------------------------------------
+| API Routes
+|--------------------------------------------------------------------------
+|
+| Here is where you can register API routes for TaskFighter.
+|
+| Assumed:
+| - prefix = api/tasks
+| - namespace = TaskFighter\Controllers
+| - middleware = api
+|
+| @see RouteServiceProvider
+|
+*/
+
+
+Route::get('list/tick', function (Response $response) {
+    $tasks = DB::table('tasks')->select('*')->get();
+    foreach ($tasks as $task) {
+        $taskFighter = new \App\Models\Task($task->name, $task->priority, $task->due_in);
+        $taskFighter->tick();
+        DB::update("update tasks set priority = '{$taskFighter->priority}', due_in = '{$taskFighter->due_in}' where id = '{$task->id}'");
```

Comment:
This needs a rewrite. I suggest having a look at Collection higher order messages support.

I think you can do this with:

``` php
Task:all()->each->tick()->each->save();
```

If that does not work and you want to do it in one loop you can do this.

``` php
Task::all()->each(function(Task $task) {
  $task->tick();
  $task->save();
});
```

**Rer:** https://laravel.com/docs/5.5/collections#higher-order-messages